components/org.wso2.carbon.identity.inbound.ldap/src/main/java/org/wso2/carbon/identity/inbound/ldap/impl/ApacheLDAPServer.java

```diff
@@ -0,0 +1,378 @@
+                    "LDAP org.wso2.carbon.identity.inbound.ldap.provider configuration is invalid.");
+            throw new IdentityLdapException("Cannot initialize LDAP org.wso2.carbon.identity.inbound.ldap.provider " +
+                    "Configuration is null");
+        }
+
+        this.endPointConfigurations = configurations;
+
+        try {
+
+            initializeDefaultDirectoryService();
+            initializeLDAPServer();
+
+        } catch (Exception e) {
+            log.error("LDAP org.wso2.carbon.identity.inbound.ldap.provider initialization failed.", e);
+            throw new IdentityLdapException("Error initializing ApacheLDAP org.wso2.carbon." +
+                    "identity.inbound.ldap.provider. ", e);
+        }
+
+    }
+
+    @Override
+    public void start()
+            throws IdentityLdapException {
+
+        try {
+            this.service.startup();
+            this.ldapServer.start();
+
+            log.info("LDAP org.wso2.carbon.identity.inbound.ldap.provider started.");
+        } catch (Exception e) {
+
+            log.error("Error starting LDAP org.wso2.carbon.identity.inbound.ldap.provider.", e);
+            throw new IdentityLdapException("Can not start the ldap org.wso2.carbon.identity.inbound.ldap.provider ", e);
+        }
+    }
+
+    @Override
+    public void stop()
+            throws IdentityLdapException {
+
+        try {
+
+            this.ldapServer.stop();
+            this.service.shutdown();
+            log.info("LDAP org.wso2.carbon.identity.inbound.ldap.provider stopped.");
+        } catch (Exception e) {
+
+            log.error("Error stopping LDAP org.wso2.carbon.identity.inbound.ldap.provider.", e);
+            throw new IdentityLdapException("Can not start the ldapendpoint", e);
+        }
+    }
+
+    protected void initializeDefaultDirectoryService()
+            throws IdentityLdapException {
+        try {
+
+            DirectoryServiceFactory factory = CarbonDirectoryServiceFactory.DEFAULT;
+            this.service = factory.getDirectoryService();
+
+            configureDirectoryService();
+
+            factory.init(this.endPointConfigurations.getInstanceId());
+
+        } catch (Exception e) {
+            throw new IdentityLdapException("Can not start the Default apacheds service ", e);
+        }
+    }
+
+    private AttributeType getAttributeType(String attributeName)
+            throws IdentityLdapException {
+        if (this.service != null) {
+            SchemaManager schemaManager = this.service.getSchemaManager();
+            if (schemaManager != null) {
+                AttributeTypeRegistry registry = schemaManager.getAttributeTypeRegistry();
+                if (registry != null) {
+                    try {
+                        String oid = registry.getOidByName(attributeName);
+                        return registry.lookup(oid);
+                    } catch (Exception e) {
```

Comment:
why r u catching Exception here? Can we not catch a specific exception?

Reply:
Here Apacheds also used the generic exception. So I had to use the same thing here.